ravendb/resource_ravendb_server.go

```diff
@@ -358,9 +655,143 @@ func parseData(d *schema.ResourceData) (ServerConfig, error) {
 		}
 	}
 
+	if d.HasChange("databases_to_delete") {
+		databasesToDeleteList := d.Get("databases_to_delete").(*schema.Set).List()
+		sc.DatabasesToDelete, err = parseDatabasesToDelete(databasesToDeleteList)
+		if err != nil {
+			return sc, err
+		}
+	}
+
+	if d.HasChange("indexes_to_delete") {
+		indexesToDeleteList := d.Get("indexes_to_delete").(*schema.Set).List()
+		sc.IndexesToDelete, err = parseIndexesToDelete(indexesToDeleteList)
+		if err != nil {
+			return sc, err
+		}
+	}
+
+	if d.HasChange("databases") {
+		databasesList := d.Get("databases").(*schema.Set).List()
+		sc.Databases, err = sc.parseDatabases(databasesList)
+		if err != nil {
+			return sc, err
+		}
+	}
+
 	return sc, nil
 }
 
+func parseIndexesToDelete(indexesToDeleteList []interface{}) ([]IndexesToDelete, error) {
+	var indexesToDelete []IndexesToDelete
+
+	for _, index := range indexesToDeleteList {
+		val := cast.ToStringMap(index)
+
+		indexesSet, err := cast.ToSliceE(val["index"])
+		if err != nil {
+			return nil, err
+		}
+
+		for _, setVal := range indexesSet {
+			index := cast.ToStringMap(setVal)
+			dbName := cast.ToString(index["database_name"])
+			indexesNamesSlice := cast.ToStringSlice(index["indexes_names"])
+			indexesToDelete = append(indexesToDelete, IndexesToDelete{
+				DatabaseName: dbName,
+				IndexesNames: indexesNamesSlice,
+			})
+		}
+	}
+
+	return indexesToDelete, nil
+}
+
+func parseDatabasesToDelete(databasesToDeleteList []interface{}) ([]DatabaseToDelete, error) {
+	var databasesToDelete []DatabaseToDelete
+	for _, database := range databasesToDeleteList {
+		val := cast.ToStringMap(database)
+
+		databases, err := cast.ToSliceE(val["database"])
+		if err != nil {
+			return nil, err
+		}
+
+		for _, db := range databases {
+			val = cast.ToStringMap(db)
+			name := cast.ToString(val["name"])
+			hardDelete := cast.ToBool(val["hard_delete"])
+
+			databasesToDelete = append(databasesToDelete, DatabaseToDelete{
+				Name:       name,
+				HardDelete: hardDelete,
+			})
+		}
+	}
+	return databasesToDelete, nil
+}
+
+func (sc *ServerConfig) parseDatabases(databasesList []interface{}) ([]Database, error) {
+	var databases []Database
+	for _, v := range databasesList {
+		val := cast.ToStringMap(v)
+		databasesSlice, err := cast.ToSliceE(val["database"])
+		if err != nil {
+			return nil, err
+		}
+
+		for _, db := range databasesSlice {
+			val = cast.ToStringMap(db)
+			name := cast.ToString(val["name"])
+			key := cast.ToString(val["encryption_key"])
+			if len(strings.TrimSpace(key)) != 0 && sc.Unsecured == true {
+				return nil, errors.New("encryption key can be used only in secured mode. ")
+			}
+			if sc.Unsecured == true {
+			}
+			databaseSettings := cast.ToStringMapString(val["settings"])
+			replicationNodes := cast.ToStringSlice(val["replication_nodes"])
+			if len(replicationNodes) == 0 {
+				replicationNodes[0] = "A"
+			}
```

Comment:
That will lead to invalid index error, no? You are verifying no elements, then set the first one?

Reply:
Correct.